### PR TITLE
First take

### PR DESCRIPTION
**Pull request by ramtina1381**

### CODE REVIEW

File: JavaScript/Timer.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    constructor(displayElement) {
+        this.display = displayElement;
+        this.state = "initial";
+        this.seconds = 0;
+        this.interval = null;
+    }
+
+    start() {
+        if (this.state === "initial") {
+            this.interval = setInterval(() => this.timer(), 1000);
```
Comment (c0d39uru) [inline JavaScript/Timer.js:17]: using 1000 here makes your timer to start almost one second after you pushed `start`.

File: JavaScript/Timer.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        let hrs = Math.floor(this.seconds / 3600);
+        let min = Math.floor((this.seconds - hrs * 3600) / 60);
+        let sec = this.seconds % 60;
+
+        if (sec < 10) sec = "0" + sec;
+        if (min < 10) min = "0" + min;
+        if (hrs < 10) hrs = "0" + hrs;
+
+        this.display.innerText = `${hrs}:${min}:${sec}`;
+        this.seconds++;
```
Comment (c0d39uru) [inline JavaScript/Timer.js]: When you pass a method to setInterval() or any other function, it is invoked with the wrong value. 
Code executed by setInterval() runs in a separate execution context than the function from which it was called. As a consequence, the this keyword for the called function is set to the window (or global) object, it is not the same as the this value for the function that called setTimeout. See [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#callbacks) for more details.

File: JavaScript/Timer.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+//this code is written to operationalize a timer with 3 main functions of Start, Stop and Reset.
```
Comment (c0d39uru) [inline JavaScript/Timer.js:1]: Use g.itignore to not include files such as .DS_Store (which is for your Mac), or `.idea` (which is for JetBrains products).

File: JavaScript/Timer.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    timer() {
+
+
+        let hrs = Math.floor(this.seconds / 3600);
+        let min = Math.floor((this.seconds - hrs * 3600) / 60);
+        let sec = this.seconds % 60;
+
+        if (sec < 10) sec = "0" + sec;
+        if (min < 10) min = "0" + min;
+        if (hrs < 10) hrs = "0" + hrs;
```
Comment (c0d39uru) [inline JavaScript/Timer.js]: You can use `padStart` instead. Also, to remove duplication, use a function here.

File: JavaScript/Timer.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+const startbtn = document.getElementById("start");
+const stopbtn = document.getElementById("stop");
+const resetbtn = document.getElementById("reset");
+
+startbtn.addEventListener("click", () => clock.start());
+stopbtn.addEventListener("click", () => clock.stop());
+resetbtn.addEventListener("click", () => clock.reset());
+
+// Initially display the timer
+clock.timer();
```
Comment (c0d39uru) [inline JavaScript/Timer.js]: This portion of the code is ugly, you are creating a bunch of `global` variables and polluting the global space. To see how problematic it is, try to show more than one stop-watch in your HTML and see if it runs smoothly. Try to manage these in a better way. Take a look at what IFEE is and how you can leverage in your case.

File: html/Timer.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<!--    fourth take-->
+
+    <meta charset="UTF-8">
+    <title>Timer</title>
+    <link rel="stylesheet" href="Timer.css">
```
Comment (c0d39uru) [inline html/Timer.html]: When I clone your code, it creates 3 directories:
- Javascript
- css
- html
In order to run your application, I had to copy the css, html, and js file in once directory and then run it.
Correct it either by moving them all in one directory or use relative addressing to load them from the right place (the latter is preferred).

File: css/Timer.css [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    font-size: 4rem;
+}
+.buttons{
+    display: flex;
+
+
+}
+#start{
+    margin: 5px;
+}
```
Comment (c0d39uru) [inline css/Timer.css:32]: Any reason you used class selector for some and ID selectors for others? (for example .button vs #start)

File: JavaScript/Timer.js (hunk 1 above)
Comment (ramtina1381) [reply JavaScript/Timer.js:17]: I called `this.timer()` right before the 14th line. It starts counting immediately after clicking the start button. But after reseting the timer and want to start it again; it takes one second to get started.

File: JavaScript/Timer.js (hunk 3 above)
Comment (ramtina1381) [reply JavaScript/Timer.js:1]: made .gitignore and passed the .idea and .DS_Store in it.

File: JavaScript/Timer.js (hunk 4 above)
Comment (ramtina1381) [reply JavaScript/Timer.js]: made a function with the name padWithZero and used the `padStart( ) ` method in it.

File: JavaScript/Timer.js (hunk 5 above)
Comment (ramtina1381) [reply JavaScript/Timer.js]: I used IIFE to pull these lines of code out of the global scope and also run them immediately.

File: html/Timer.html (hunk 6 above)
Comment (ramtina1381) [reply html/Timer.html]: Sorry about that! I fixed the relative path. I made the  separate directories after writing my code, and forgot to modify the relative paths.

File: css/Timer.css (hunk 7 above)
Comment (ramtina1381) [reply css/Timer.css:32]: I thought maybe I would have some other buttons later on and I would possibly like to keep the css same for them. but start, stop and reset buttons are very specific and they will not be used anywhere else probably.

**Pull request merged**